Zend/zend_builtin_functions.c

```diff
@@ -1587,14 +1639,44 @@ static void debug_backtrace_get_args(zend_execute_data *call, zval *arg_array) /
 			}
 
 			while (i < num_args) {
+				zval arg;
+
+				zend_attribute *attribute = zend_get_parameter_attribute_str(
+					call->func->op_array.attributes,
+					"sensitiveparameter",
+					sizeof("sensitiveparameter") - 1,
+					i
+				);
+
+				if (
+					(i > call->func->op_array.num_args && last_was_sensitive)
```

Comment:
This is a bit cryptic. Took me a while to understand.

* `zend_get_parameter_attribute_str` could be skipped if `i > call->func->op_array.num_args` since that will always be `NULL`
* The check should be `i >= call->func->op_array.num_args` but it's used as a side-effect to make the first superfluous argument set the `last_was_sensitive` to `0`, in combination with the fact that the variadic parameter doesn't count towards `op_array.num_args` to keep it at `1`.

```c
// The first argument after num_args can still have an attribute due to variadics, thus <=
if (i <= call->func->op_array.num_args) {
	zend_attribute *attribute = zend_get_parameter_attribute_str(
		call->func->op_array.attributes,
		"sensitiveparameter",
		sizeof("sensitiveparameter") - 1,
		i
	);
	last_was_sensitive = attribute != NULL;
}
```

Seems a bit more explicit.

Also, superfluous arguments (without variadics) are missing a test.

```php
function test(
    $non_sensitive,
    #[SensitiveParameter] $sensitive,
)
{
    debug_print_backtrace();
    var_dump(debug_backtrace());
    var_dump((new Exception)->getTrace());
}

test('foo', 'bar', 'baz');
```

Reply:
> Also, superfluous arguments (without variadics) are missing a test.

Thanks, added.

Regarding the code suggestion: I'll look into that later.